src/main/java/com/aws/greengrass/deployment/DeviceConfiguration.java

```diff
@@ -971,4 +986,12 @@ public KeyManager[] getDeviceIdentityKeyManagers() throws TLSAuthException {
     public Topics getHttpClientOptions() {
         return getTopics("httpClient");
     }
+
+    /**
+     * Get configuration from kernel.
+     *
+     */
+    public Configuration getConfig() {
+        return kernel.getConfig();
+    }
```

Comment:
Remove this. If you want the configuration, ask the kernel for the configuration, not DeviceConfiguration